### DISCUSSION TITLE
How to use Meshroom in my python project without GUI?

### DISCUSSION THREAD

**Discussion by AD288** [alicevision/Meshroom#1775] [Q&A]
Hello I found that we can do it using (https://github.com/alicevision/Meshroom/blob/develop/bin/meshroom_batch) , but I’m junior in python, what’s the procedure? how can I launch with my images and get the results using this file?
Thanks

**Comment by GreatScott** [comment]
Hi @AD288,

Bit of a meshroom noob myself, but I noticed you can see the command line calls for each stage of the pipeline under the "status" tab for each node (see lower-right corner of the attached screenshot). As a first step, I would see if you can recreate all the calls just in your OS's terminal. Once you are able to reliably do that, you can then have Python make the calls by using subprocess module.

https://realpython.com/python-subprocess/

![image](https://user-images.githubusercontent.com/1652604/194176306-66c1bf6b-c376-432a-ad9a-6e194e2130a8.png)

Hope this helps!

**Reply by natowi** [reply]
That is correct, Meshroom is a GUI (python/qt based) that calls AliceVision with cli parameters. You can do the same.
